### PR TITLE
Sonar Branch Analysis

### LINKED ISSUE #190: Add Other Branch Analysis to SonarQube

### PR DESCRIPTION
**Pull request by grylajoie**
## What does this PR do?
Small addition to the CI pipeline to add CI analysis instead of Automatic for SonarQube. This enables us to upload Test Coverage to SonarCloud too.

Note: Other branch analysis such as our dev branch cannot be analyzed due to it being a paid feature.

## Linked issue
Closes #190 

## Review checklist
- [ ] code is readable
- [ ] tests updated/added
- [ ] edge cases handled

### CODE REVIEW

**Comment by grylajoie** [comment]
This PR only has changed in CI pipeline, no conflicts with app. @lucaieraci check it quick

File: mobile/sonar-project.properties [hunk 1]
```diff
@@ -0,0 +1,15 @@
+sonar.projectKey=HotfixInProduction_hotfixinprod
+sonar.organization=hotfixinproduction
+
+# Path to source directories
+sonar.sources=src
+sonar.tests=__tests__
+
+# Exclude coverage files and test files
+sonar.exclusions=**/*.test.tsx,**/*.test.ts,**/coverage/**,**/node_modules/**,**/android/**,**/ios/**,**/__mocks__/**,**/package-lock.json,**/yarn.lock,**/pnpm-lock.yaml,**/*.config.js,**/*.config.ts,**/dist/**,**/build/**,**/.expo/**,**/.expo-shared/**,**/assets/**,**/*.json,**/gradlew,**/gradlew.bat,**/*.gradle,**/*.properties
```
Comment (lucaieraci) [inline mobile/sonar-project.properties]: This variable is quite long can we do some refactoring to shorten it or change it to something else.

Comment (grylajoie) [reply mobile/sonar-project.properties]: @lucaieraci Can you check again and approve it?

Comment (lucaieraci) [reply mobile/sonar-project.properties]: OK looks better
